bindings.c

```diff
@@ -3794,6 +3799,90 @@ static uint64_t get_reaper_age(pid_t pid)
 	return procage;
 }
 
+/*
+ * Returns 0 on success.
+ * It is the caller's responsibility to free `return_usage`, unless this
+ * function returns an error.
+ */
+static int read_cpuacct_usage_all(char *cg, char *cpuset, struct cpuacct_usage **return_usage)
+{
+
+	int cpucount = get_nprocs();
+	struct cpuacct_usage *cpu_usage;
+	int rv = 0, i, j, ret, read_pos = 0, read_cnt;
+	int cg_cpu;
+	uint64_t cg_user, cg_system;
+	int64_t ticks_per_sec;
+	char *usage_str = NULL;
+
+	ticks_per_sec = sysconf(_SC_CLK_TCK);
+
+	if (ticks_per_sec < 0 && errno == EINVAL) {
+		lxcfs_debug(
+			"%s\n",
+			"read_cpuacct_usage_all failed to determine number of clock ticks "
+			"in a second");
+		return -1;
+	}
+
+	cpu_usage = malloc(sizeof(struct cpuacct_usage) * cpucount);
+	if (!cpu_usage)
+		return -ENOMEM;
+
+	if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_all", &usage_str)) {
+		rv = -1;
+		goto err;
+	}
+
+	if (sscanf(usage_str, "cpu user system\n%n", &read_cnt) != 0) {
+		lxcfs_error("read_cpuacct_usage_all reading first line from "
+				"%s/cpuacct.usage_all failed.\n", cg);
+		rv = -1;
+		goto err;
+	}
+
+	read_pos += read_cnt;
+
+	for (i = 0, j = 0; i < cpucount; i++) {
+		ret = sscanf(usage_str + read_pos, "%d %lu %lu\n%n", &cg_cpu, &cg_user,
+				&cg_system, &read_cnt);
+
+		if (ret == EOF) {
+			break;
+
+		} else if (ret != 3) {
```

Comment:
Nit: There's no need for the `if else` ladder here. You can just do:
```
if (ret == EOF)
        break;

if (ret != 3) {
.
.
.
```

Reply:
Fixed